tests/test_merge.py

```diff
@@ -379,6 +381,332 @@ def test_merge(
+            raises(ValidationError, match="child 'foo' is not Optional"),
+            None,
+            None,
+            id="new_user_none",
+        ),
+        param(
+            (DictConfig(content={}, element_type=User), {"foo": MISSING}),
+            {"foo": MISSING},
+            User,
+            False,
+            id="new_user_missing",
+        ),
+        param(
+            (
+                DictConfig(content={}, element_type=Optional[User]),
+                {"foo": User("Bond")},
+            ),
+            {"foo": User("Bond")},
+            User,
+            True,
+            id="new_optional_user",
+        ),
+        param(
+            (DictConfig(content={}, element_type=Optional[User]), {"foo": None}),
+            {"foo": None},
+            User,
+            True,
+            id="new_optional_user_none",
+        ),
+        param(
+            (DictConfig(content={}, element_type=Optional[User]), {"foo": MISSING}),
+            {"foo": MISSING},
+            User,
+            True,
+            id="new_optional_user_missing",
+        ),
+        param(
+            (
+                DictConfig(content={"foo": MISSING}, element_type=User),
+                {"foo": User("Bond")},
+            ),
+            {"foo": User("Bond")},
+            User,
+            False,
+            id="missing_user",
+        ),
+        param(
+            (DictConfig(content={"foo": MISSING}, element_type=User), {"foo": None}),
+            raises(ValidationError, match="field 'foo' is not Optional"),
+            None,
+            None,
+            id="missing_user_none",
+        ),
+        param(
+            (DictConfig(content={"foo": MISSING}, element_type=User), {"foo": MISSING}),
+            {"foo": MISSING},
+            User,
+            False,
+            id="missing_user_missing",
+        ),
+        param(
+            (
+                DictConfig(content={"foo": MISSING}, element_type=Optional[User]),
+                {"foo": User("Bond")},
+            ),
+            {"foo": User("Bond")},
+            User,
+            True,
+            id="missing_optional_user",
+        ),
+        param(
+            (
+                DictConfig(content={"foo": MISSING}, element_type=Optional[User]),
+                {"foo": None},
+            ),
+            {"foo": None},
+            User,
+            True,
+            id="missing_optional_user_none",
+        ),
+        param(
+            (
+                DictConfig(content={"foo": MISSING}, element_type=Optional[User]),
+                {"foo": MISSING},
+            ),
+            {"foo": MISSING},
+            User,
+            True,
+            id="missing_optional_user_missing",
+        ),
+    ],
+)
+def test_optional_element_type_merge(
+    inputs: Any, expected: Any, ref_type: Any, is_optional: bool
+) -> None:
+    configs = [_ensure_container(c) for c in inputs]
+    if isinstance(expected, AbstractContextManager):
```

Comment:
Other tests typically use `RaisesContext` in this situation, any reason not to do the same?

Reply:
Oh, I hadn't realized that other tests in this repo are using `RaisesContext`.
Updated in https://github.com/omry/omegaconf/pull/749/commits/1ab08a6b51dd6f33296c76c589e991dfde32cc0c.